File: 2A/ValueVsReferenceCSharp/ValueVsReferenceCSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ValueVsReferenceCSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            int number;
            number = 1111111;
            Overwirite(number);
            this.richTextBox1.AppendText(number.ToString() + "\n");
        }


        private void button2_Click(object sender, EventArgs e)
        {
            Student a = new Student();
            a.name = "Pippo";
            Overwirite(a);
            this.richTextBox1.AppendText(a.name + "\n");
        }


        private void Overwirite(int var)
        {
            var = 999999999;
        }

        static void Overwirite(Student var)
        {
            var.name = "9999999";
        }

        private class Student
        {
            public string name;
        }
    }
}

File: 2A/ValueVsReferenceVB/ValueVsReferenceVB/Form1.vb
Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim number As Integer
        number = 1111111
        Overwirite(number)
        Me.RichTextBox1.AppendText(number.ToString() & vbLf)
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Dim a As Student = New Student()
        a.name = "Pippo"
        Overwirite(a)
        Me.RichTextBox1.AppendText(a.name & vbLf)
    End Sub

    Private Sub Overwirite(ByVal var As Integer)
        var = 999999999
    End Sub

    Private Shared Sub Overwirite(ByVal var As Student)
        var.name = "9999999"
    End Sub
    Private Class Student
        Public name As String
    End Class


End Class

